projects/habitat_ovmm/receptacles_data_collection.py

```diff
@@ -0,0 +1,267 @@
+            config.habitat.gym.obs_keys.remove("robot_third_rgb")
+        if "third_rgb_sensor" in config.habitat.simulator.agents.main_agent.sim_sensors:
+            config.habitat.simulator.agents.main_agent.sim_sensors.pop(
+                "third_rgb_sensor"
+            )
+
+    episode_ids_range = config.habitat.dataset.episode_indices_range
+    if episode_ids_range is not None:
+        config.EXP_NAME = os.path.join(
+            config.EXP_NAME, f"{episode_ids_range[0]}_{episode_ids_range[1]}"
+        )
+
+    OmegaConf.set_readonly(config, True)
+    return config
+
+def receptacle_position_aggregate(env):
+    # This is for iterating through all episodes once using only one env
+    count_dict = {}
+    for episode in env._dataset.episodes:
+        scene_id = extract_scene_id(episode.scene_id)
+        hash_str = f'ep_{episode.episode_id}_scene_{scene_id}'
+        count_dict[hash_str] = 0
+        
+    receptacle_positions = dict()    
+
+    # Ideally, we can make it like an iterator to make it feel more intuitive
+    while True:
+        # Get a new episode
+        obs = env.reset()
+        episode = env.get_current_episode()
+        scene_id = extract_scene_id(episode.scene_id)
+
+        # Check if you have iterated through all episodes and if yes, break the loop
+        hash_str = f'ep_{episode.episode_id}_scene_{scene_id}'
+        if count_dict[hash_str] == 0:
+            count_dict[hash_str] += 1
+        elif count_dict[hash_str] == 1:
+            break
+        else:
+            raise NotImplementedError # Technically, this shouldn't happen
+
+        if not scene_id in receptacle_positions:
+            receptacle_positions[scene_id] = dict()
+
+        if not episode.goal_recep_category in receptacle_positions[scene_id]:
+            receptacle_positions[scene_id][episode.goal_recep_category] = set() 
+        for recep in episode.candidate_goal_receps:
+            recep_position = list(recep.position)
+            view_point_position = list(recep.view_points[0].agent_state.position)
+            receptacle_positions[scene_id][episode.goal_recep_category].add(tuple(recep_position + view_point_position))    
+
+    os.makedirs('./scene_info', exist_ok=True)
+    with open('./scene_info/recep_position.pickle', 'wb') as handle:
+        pickle.dump(receptacle_positions, handle, protocol=pickle.HIGHEST_PROTOCOL)
+    return receptacle_positions
+
+def gen_receptacle_images(env):
+    sim = env.habitat_env.env._env._env._sim 
+    
+    # This is for iterating through all episodes once using only one env
+    # Also, creating folders for storing dataset
+    count_dict = {}
+    os.makedirs('./data_out', exist_ok=True)
+    for episode in env._dataset.episodes:
+        scene_id = extract_scene_id(episode.scene_id)
+        os.makedirs(f'./data_out/scene_{scene_id}', exist_ok=True)
+        hash_str = f'ep_{episode.episode_id}_scene_{scene_id}'
+        count_dict[hash_str] = 0
```

Comment:
Is this code duplicated from last one? could you make it a helper function

Reply:
created a helped function to avoid duplication